.buildkite/ftr_configs.yml

```diff
@@ -437,3 +440,9 @@ enabled:
   - x-pack/performance/journeys/apm_service_inventory.ts
   - x-pack/test/custom_branding/config.ts
   - x-pack/test/profiling_api_integration/cloud/config.ts
+  - x-pack/test/security_solution_api_integration/test_suites/detections_response/serverless.config.ts
+  - x-pack/test/security_solution_api_integration/test_suites/detections_response/ess.config.ts
```

Comment:
Having only two configs for all D&R tests will not scale, unfortunately. We need to split our tests into N groups, where each group will have at least two FTR configs: one for ESS and one for Serverless. In the near future, groups will also have additional FTR configs for enabling feature flags and running tests against enabled features. Please read more about the idea of splitting tests into meaningful groups in https://github.com/elastic/kibana/issues/151902.

Also, there's a requirement that each group of FTR tests must run under 40 minutes. This means we should aim at 15-20 minutes at most per group, leaving some time buffer for incidents where CI gets slower than it normally is.

All that can force us to split tests into more or less fine-grained groups, sometimes more fine-grained than our subdomains (exceptions, rule management, etc). I'd say that we will likely have multiple groups per subdomain.

Let's figure out what groups should be created for the tests moved in this PR.

Reply:
I've planned to work on https://github.com/elastic/kibana/issues/151902 after completing this PR. This subsequent issue involves some research to determine how we can efficiently group tests based on their relevance and execution time. 

The idea behind this PR is to establish the main structural changes first, reducing the number of files and scoping this PR accordingly. 

Also approaching the grouping in a separate PR  will serve in documenting our approach to grouping tests, making it easier for reviewers to understand. hopefully, this separation of tasks helps streamline the review process and ensures that our tests are organized and well-documented.

I am open to discussing this topic with the whole team